fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerConflictsTest.java

```diff
@@ -201,5 +201,60 @@ public void run() {
         assertSame(recordStore.getIndexState(index.getName()), IndexState.READABLE);
         assertEquals(0, timer.getCount(FDBStoreTimer.Events.COMMIT_FAILURE));
     }
+
+    @Test
+    public void testOnlineIndexerIdempotentNested() throws Exception {
+        // If a conflict happens, this test should succeed by retries
+        Random rand = new Random();
+        final long seed = rand.nextInt();
+        LOGGER.info("Random seed is " + seed);
+        rand = new Random(seed); // make rand repeatable (note that multi threading behaviour is not fully repeatable in this test)
+
+        List<TestRecords1Proto.MySimpleRecord> records =
+                LongStream.range(0, 400).mapToObj(val -> TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(val).setNumValue2((int)val + 1).build()
+                ).collect(Collectors.toList());
```

Comment:
I don't think this test needs quite as many records now. I think...three would suffice? The main tests have lots of records in order to make sure that during the random operations, we get lots of different combinations. That shouldn't be necessary with this test, though, as it's more about testing specific concurrent operations.

Reply:
Sure. Shall I keep the other two (random) tests?

Reply:
I don't think so, really. I think the tests in `OnlineIndexerBuildValueIndexTest` and `OnlineIndexerBuildSumIndexTest` are essentially trying to test the same thing, and so I think we'd want to stick with those (or add more tests there if we think they are insufficient).

I'd also think we'd want to track down the weird conflict if we kept them in.

Reply:
Ok.